### PR TITLE
Support multiple outputs in the converter

### PR DESCRIPTION
**Pull request by yanndupis**
Hello, 

Currently, the converter doesn't support ops with multiple outputs when the op is not registered as a special op. Here is an attempt to fix this. By doing so, it should fix the issue for the `tf.split` operation identified in this pr #421.

Thanks,

### CODE REVIEW

File: tf_encrypted/convert/convert.py [hunk 1]
```diff
@@ -95,6 +95,18 @@ def convert(
           continue
 
         self.outputs[output] = register[node.op](self, node, inputs)
```
Comment (jvmancuso) [inline tf_encrypted/convert/convert.py]: Shouldn't this line be deleted now?

File: tf_encrypted/convert/convert.py [hunk 2]
```diff
@@ -83,7 +83,7 @@ def convert(
       if node.name not in specop_outputs:
 
         output = strip_tensor_info(node.name)
-        inputs = [strip_tensor_info(x) for x in node.input]
+        inputs = [x for x in node.input]
```
Comment (jvmancuso) [inline tf_encrypted/convert/convert.py]: why this change?

File: tf_encrypted/convert/convert.py (hunk 1 above)
Comment (yanndupis) [reply tf_encrypted/convert/convert.py]: Ah yes, my bad, it should have been removed.

File: tf_encrypted/convert/convert.py (hunk 2 above)
Comment (yanndupis) [reply tf_encrypted/convert/convert.py]: ah it's because `strip_tensor_info` removes everything after `:` in the node name. However, at least for `split` operation, the outputs are distinguished by what's after `:`. For example, for split (in 3), it will be `split`, `split:1`, `split:2`. If we remove everything after `:`, the first output will be re-used three times.

Comment (jvmancuso) [reply tf_encrypted/convert/convert.py]: awesome, thanks for clarifying!

File: tf_encrypted/convert/convert.py [hunk 3]
```diff
@@ -94,7 +94,17 @@ def convert(
           self.outputs[output] = x
           continue
 
-        self.outputs[output] = register[node.op](self, node, inputs)
+        out = register[node.op](self, node, inputs)
+
+        # if the operation returns a list with several ouputs,
+        # identify the outputs node name
+        if isinstance(out, list):
```
Comment (jvmancuso) [inline tf_encrypted/convert/convert.py]: can we switch this to `isinstance(out, (list, tuple))`? feels a bit more robust

File: tf_encrypted/convert/convert.py [hunk 4]
```diff
@@ -293,3 +303,33 @@ def strip_tensor_info(node_name: str) -> str:
...
+
+  node {
+    name: "output/input"
+    op: "Pack"
+    input: "split"
+    input: "split:1"
+    }
+  """
+  output_node = []
+  for n in pb_trimmed:
```
Comment (jvmancuso) [inline tf_encrypted/convert/convert.py]: wish we could do this without iterating through the entire protobuf. I feel like we should be doing a forward lookahead from the current node that breaks when it stops encountering tensors that are outputs of the current node)

Comment (jvmancuso) [reply tf_encrypted/convert/convert.py]: this would assume that the nodes are ordered topologically, but if so that would be quite a bit faster

File: tf_encrypted/convert/convert.py (hunk 3 above)
Comment (yanndupis) [reply tf_encrypted/convert/convert.py]: definitely, good point.

File: tf_encrypted/convert/convert.py (hunk 4 above)
Comment (yanndupis) [reply tf_encrypted/convert/convert.py]: agreed, good point.

Comment (yanndupis) [reply tf_encrypted/convert/convert.py]: I tweaked the function so it doesn't go through the entire pb file. Let me know what you think. Thanks.
